test/unit/finance/bill_variable_for_plan_changed_test.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Finance::BillVariableForPlanChangedTest < ActiveSupport::TestCase
+  attr_reader :contract, :app_plan
+
+  setup do
+    @contract = FactoryBot.build_stubbed(:contract)
+    account = FactoryBot.build_stubbed(:simple_account)
+    @app_plan = FactoryBot.build_stubbed(:application_plan)
+
+    contract.stubs(:provider_account).returns(account)
+    account.stubs(:provider_can_use?).with(:instant_bill_plan_change).returns(true)
+
+    ENV["TZ"] = "UTC"
```

Comment:
Wait, I didn't do that, I submitted a new PR with my fixes to `no_variable_cost_test.rb` - #2837 

I see @josemigallas added to this PR a number of other changes that should better be reviewed separately from my changes.

Reply:
Also it will be much easier if we split logic from style changes in different PRs to make reviews easier. Need to read line by line to see what are the differences.